### PR TITLE
[Spark] Fix sparkjob resources to be handled correctly with K8S configuration

### PR DESCRIPTION
**Pull request by benbd86**
Making limits handle cpu mem and gpu and requests handle mem and cpu to be aligned with k8s config

[ML-2136](https://jira.iguazeng.com/browse/ML-2136)
[ML-2142](https://jira.iguazeng.com/browse/ML-2142)

### CODE REVIEW

File: mlrun/runtimes/sparkjob/abstract.py [hunk 1]
```diff
@@ -231,18 +230,20 @@ def deploy_default_image(cls, with_gpu=False):
         sj = new_function(kind=cls.kind, name="spark-default-image-deploy-temp")
         sj.spec.build.image = cls._get_default_deployed_mlrun_image_name(with_gpu)
 
-        sj.with_executor_requests(cpu=1, mem="512m", gpus=1 if with_gpu else None)
-        sj.with_driver_requests(cpu=1, mem="512m", gpus=1 if with_gpu else None)
+        sj.with_executor_requests(cpu=1, mem="512m")
```
Comment (gtopper) [inline mlrun/runtimes/sparkjob/abstract.py:233]: We don't want to request a GPU anymore?

File: mlrun/runtimes/sparkjob/abstract.py [hunk 2]
```diff
@@ -231,18 +230,20 @@ def deploy_default_image(cls, with_gpu=False):
...
+        sj.with_executor_limits(cpu=1, mem="512m", gpus=1 if with_gpu else None)
+        sj.with_driver_requests(cpu=1, mem="512m")
```
Comment (gtopper) [inline mlrun/runtimes/sparkjob/abstract.py:235]: We don't want to request a GPU anymore?

File: tests/api/runtimes/test_spark.py [hunk 3]
```diff
@@ -115,6 +118,28 @@ def _assert_volume_and_mounts(
                 == {}
             )
 
+    def _assert_driver_and_executor(self, body, expected_values):
```
Comment (gtopper) [inline tests/api/runtimes/test_spark.py:121]: Should probaby be named `_assert_driver_and_executor_resources` or something like that.

**Review by gtopper — APPROVED** [review]

File: mlrun/runtimes/sparkjob/abstract.py [hunk 4]
```diff
@@ -231,18 +230,20 @@ def deploy_default_image(cls, with_gpu=False):
...
+        sj.with_driver_limits(cpu=1, mem="512m", gpus=1 if with_gpu else None)
```
Comment (Hedingber) [inline mlrun/runtimes/sparkjob/abstract.py:236]: This function is just here to build the image, I'm not sure why it even needs requests or limits set

**Review by Hedingber — CHANGES_REQUESTED** [review]

**Comment by oritnm** [comment]
let's verify that a different values can be set on memory ( limit/request) for spark application jobs. 
see https://github.com/apache/spark/blob/95c7efd7571464d8adfb76fb22e47a5816cf73fb/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/BasicDriverFeatureStep.scala#L55

**Pull request merged**